### DISCUSSION TITLE
Line plots of 3D data

### DISCUSSION THREAD

**Discussion by ZedThree** [pydata/xarray#6365] [Q&A] [answered]
I'm trying to recreate an existing plot of a 3D array:

```python
for i in range(array.shape[1]):
    for j in range(array.shape[2]):
        plt.plot(time, array[:, i, j])
```

but the naive translation `array.plot.line(x="t")` gives me

```
ValueError: Line plots are for 1- or 2-dimensional DataArrays. Passed DataArray has 3 dimensions
```

Sort of dropping that restriction from xarray itself, what's the most idiomatic way to do this with xarray? Is there a nice way to iterate over a particular coordinate? Obviously the original method still works, but it doesn't feel very idiomatic!

**Comment by kmuehlbauer** [comment]
Maybe just select the wanted entries, something along the line (dimension names have to be changed according to your array):

``array.isel(dim1=i, dim2=j).plot()``

**Reply by ZedThree** [reply]
Possibly a stupid question, but is there any difference between `array[:, i, j]` and `array.isel(dim1=i, dim2=j)`?

**Reply by kmuehlbauer** [reply]
They are interchangeable AFAIK, but maybe I misunderstood your question.

**Reply by TomNicholas** [reply]
The only differences are that:
- The `isel` call will work regardless of the dimensionality of your array, whereas the array indexing expects that your array is 3D,
- The `isel` call requires that your array have dimensions named `'dim1'` and `'dim2'`, whereas the array indexing doesn't care what the names of your dimensions are, because its just looking up nameless numpy axes,
- The `isel` call is insensitive to the transposition of your array (and so `array.isel(dim1=i, dim2=j)` is equivalent too).

So as long as `array` is 3D, has a `dim1` and a `dim2`, which are the 2nd and 3rd dimensions, then the calls should be equivalent.

**Comment by TomNicholas** [comment, marked as answer]
You could maybe also try using [`stack`](https://docs.xarray.dev/en/latest/generated/xarray.DataArray.stack.html) to stack the two dimensions you want to loop over into one dimension, before plotting?
